src/operations/execute_operation.ts

```diff
@@ -86,6 +86,8 @@ export function executeOperation<
         session = topology.startSession({ owner, explicit: false });
       } else if (session.hasEnded) {
         return cb(new MongoDriverError('Use of expired sessions is not permitted'));
+      } else if (session.snapshotEnabled && !topology.capabilities.supportsSnapshotReads) {
+        return cb(new MongoDriverError('Snapshot reads require MongoDB 5.0 or later'));
```

Comment:
Do we want to use a more specific error here now that they are available?

Reply:
Compat and InvalidArgument are still not available so I think leaving the DriverError in there would make sense for now.